src/action.js

```diff
@@ -0,0 +1,182 @@
+import * as os from "os";
+
+import * as github from "@actions/github";
+import * as toolCache from "@actions/tool-cache";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+
+// Define consts
+
+export const gruntworkOrg = "gruntwork-io";
+export const patcherRepo = "patcher-cli";
+export const patcherVersion = "v0.4.3";
+
+export const reportCommand = "report";
+export const updateCommand = "update";
+
+export const nonInteractiveFlag = "--non-interactive"
+export const noColorFlag = "--no-color"
+export const skipContainerFlag = "--skip-container-runtime"
+
+function osPlatform() {
+    switch (os.platform()) {
+        case "linux":
+            return "linux";
+        case "darwin":
+            return "darwin";
+        default:
+            core.setFailed("Unsupported operating system - the Patcher action is only released for Darwin and Linux");
+            return;
+    }
+}
+
+export async function openPullRequest(output, dependency, ghToken) {
+    const head = `patcher-updates-${dependency}`
+    const title = `[Patcher] Update ${dependency}`
+    const commitMessage = "Update dependencies using Patcher"
+    const commitAuthor = "Grunty"
+    const commitEmail = "grunty@gruntwork.io"
+
+    const body = `
+Updated the \`${dependency}\` dependency using Patcher.
+
+### Update summary
+\`\`\`yaml
+${output}
+\`\`\`
+`
+
+    await exec.exec("git", ["config", "user.name", commitAuthor])
+    await exec.exec("git", ["config", "user.email", commitEmail])
+    await exec.exec("git", ["add", "."])
+    await exec.exec("git", ["commit", "-m", commitMessage])
+    await exec.exec("git", ["checkout", "-b", head])
+
+    const context = github.context;
+
+    await exec.exec("git", ["push", "-f", `https://${ghToken}@github.com/${context.repo.owner}/${context.repo.repo}.git`])
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const repoDetails = await octokit.rest.repos.get({ ...context.repo });
+    const base = repoDetails.data.default_branch;
+    core.debug(`Base branch is ${base}. Opening the PR against it.`)
+
+    try {
+        await octokit.rest.pulls.create({ ...context.repo, title, head, base, body, });
+    } catch (error) {
+        if (error.message?.includes(`A pull request already exists for`)) {
+            core.error(`A pull request for ${head} already exists. The branch was updated.`)
+        } else {
+            throw error
+        }
+    }
+}
+
+export async function downloadRelease(owner, repo, tag, ghToken) {
+    core.info(`Downloading Patcher version ${patcherVersion}`);
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const getReleaseUrl = await octokit.rest.repos.getReleaseByTag({ owner, repo, tag })
+
+    const re = new RegExp(`${osPlatform()}.*amd64`)
+    let asset = getReleaseUrl.data.assets.find(obj => {
```

Comment:
`find(...)` could return `undefined` and it doesn't look like you are handling that.

Also `let` should be a `const` since you're not re-assigning `asset` after you initialize it here.